x-pack/plugins/maps/public/shared/layers/styles/vector_style.js

```diff
@@ -157,7 +178,9 @@ export class VectorStyle extends AbstractStyle {
     }
 
     const featuresMeta = {
```

Comment:
consider using the actual VECTOR_FEATURE_TYPE enumeration values as keys in the `styleMeta` object.

So not  {hasLines, hasPolygon, hasPoints} but 

```
{
  hasFeatureType: {
      [VECTOR_FEATURE_TYPE.POINT]: [true|false], 
      [VECTOR_FEATURE_TYPE.LINE]: [true|false],
      [VECTOR_FEATURE_TYPE.POLYGON]: [true|false]
    }
}
```